eip712/src/main/java/pm/gnosis/eip712/EIP712.kt

```diff
@@ -0,0 +1,51 @@
+package pm.gnosis.eip712
+
+import pm.gnosis.crypto.utils.Sha3Utils
+import pm.gnosis.model.Solidity
+import pm.gnosis.model.SolidityBase
+import pm.gnosis.utils.hexToByteArray
+
+val solidityTypes = Solidity.types.keys
+const val DOMAIN_TYPE = "EIP712Domain"
+
+sealed class Type712 {
+    abstract val type: String
+    abstract fun encodeData(): ByteArray
+}
+
+class Literal712(val value: SolidityBase.Type) : Type712() {
+    override fun encodeData(): ByteArray = when (value) {
+        // Solidity.String is also Solidity.Bytes
+        is Solidity.Bytes -> Sha3Utils.keccak(value.items)
+        is SolidityBase.Array<*> -> value.items.map { Literal712(value = it).encodeData() }.reduce { acc, bytes -> acc + bytes }
+        is SolidityBase.Vector<*> -> value.items.map { Literal712(value = it).encodeData() }.reduce { acc, bytes -> acc + bytes }
```

Comment:
if `encodeData()` would be a method that takes a SolidityBase.Type then we wouldn't need to create a `Literal712` here should for encoding purpose

Reply:
Fixed in https://github.com/gnosis/svalinn-kotlin/pull/31/commits/4a12c6ac72088ff9125cea04910c5a5f74878570